tests/TestFileWriterRotationScenarios/tst_FileWriterRotations.cpp

```diff
@@ -959,45 +977,50 @@ void FileWriterRotations::testRotateForStrictNumbers(int compressionAlgorithm)
     m_Config.compressionAlgo = compressionAlgorithm;
 
     QString compressedExt = getCompressedExten(compressionAlgorithm);
+    //in this test we use a subfolder
+    QDir::current().mkdir("test_subfolder");
+    QString path = QDir::currentPath() + "/test_subfolder";
 
     //create base file
-    setOutputFile(filenames[0]);
+    setOutputFile(path + filenames[0]);
 
     //write to log.txt more than allowed size and rotate
-    writeToFile(filenames[0], 2);
+    writeToFile(path + filenames[0], 2);
     rotateFileForStrictRotation();
 
     //verify we created the new file
-    QVERIFY(QFileInfo::exists(filenames[0]));
-    QVERIFY(QFileInfo::exists(filenames[1] + compressedExt));
-    QVERIFY(getCurrentLogFilename() == filenames[0]);
+    QVERIFY(QFileInfo::exists(path + filenames[0]));
+    QVERIFY(QFileInfo::exists(path + filenames[1] + compressedExt));
+    QVERIFY(getCurrentLogFilename() == getFileWithFullPath(filenames[0], path));
 
 
     //write to log more than allowed size and rotate
-    writeToFile(filenames[0], 2);
+    writeToFile(path + filenames[0], 2);
     rotateFileForStrictRotation();
 
-    QVERIFY(QFileInfo::exists(filenames[0]));
-    QVERIFY(QFileInfo::exists(filenames[1] + compressedExt));
-    QVERIFY(QFileInfo::exists(filenames[2] + compressedExt));
-    QVERIFY(!QFileInfo::exists(filenames[3] + compressedExt));
-    QVERIFY(getCurrentLogFilename() == filenames[0]);
+    QVERIFY(QFileInfo::exists(path + filenames[0]));
+    QVERIFY(QFileInfo::exists(path + filenames[1] + compressedExt));
+    QVERIFY(QFileInfo::exists(path + filenames[2] + compressedExt));
+    QVERIFY(!QFileInfo::exists(path + filenames[3] + compressedExt));
+    QVERIFY(getCurrentLogFilename() == getFileWithFullPath(filenames[0], path));
 
 
     //write to log more than allowed size and rotate
-    writeToFile(filenames[0], 2);
+    writeToFile(path + filenames[0], 2);
     rotateFileForStrictRotation();
 
-    QVERIFY(!QFileInfo::exists(filenames[3] + compressedExt));
-    QVERIFY(QFileInfo::exists(filenames[2] + compressedExt));
-    QVERIFY(QFileInfo::exists(filenames[1] + compressedExt));
-    QVERIFY(QFileInfo::exists(filenames[0]));
-    QVERIFY(getCurrentLogFilename() == filenames[0]);
+    QVERIFY(!QFileInfo::exists(path + filenames[3] + compressedExt));
+    QVERIFY(QFileInfo::exists(path + filenames[2] + compressedExt));
+    QVERIFY(QFileInfo::exists(path + filenames[1] + compressedExt));
+    QVERIFY(QFileInfo::exists(path + filenames[0]));
+    QVERIFY(getCurrentLogFilename() == getFileWithFullPath(filenames[0], path));
 
     foreach (QString s, filenames) {
-        deleteFile(s + compressedExt);
+        deleteFile(path + s + compressedExt);
```

Comment:
Also here and in all the previous line, use `getFileWithFullPath` instead of `path + ...`